pkg/api/controller.go

```diff
@@ -3025,38 +3077,35 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
-func (c *Controller) sendResetPasswordEmail(email string, token string) error {
-	return c.Emailer.SendEmailWithLimit([]string{email}, token, token, nil)
-}
-
-func (c *Controller) ForgotPassword(w http.ResponseWriter, r *http.Request, body ForgotPasswordJSONRequestBody) {
-	addr, err := mail.ParseAddress(body.Email)
+func (c *Controller) resetPasswordRequest(ctx context.Context, email string) error {
+	addr, err := mail.ParseAddress(email)
 	if err != nil {
-		c.Logger.WithError(err).WithField("email", body.Email).Debug("forgot password with invalid email")
-		w.WriteHeader(http.StatusNoContent)
-		return
+		return err
 	}
-	user, err := c.Auth.GetUserByEmail(r.Context(), addr.Address)
+	user, err := c.Auth.GetUserByEmail(ctx, addr.Address)
 	if err != nil {
-		c.Logger.WithError(err).WithField("email", addr.Address).Debug("failed to retrieve user by email")
-		w.WriteHeader(http.StatusNoContent)
-		return
+		return err
 	}
-	email := StringValue(user.Email)
-	secret := c.Auth.SecretStore().SharedSecret()
-	currentTime := time.Now()
-	token, err := GenerateJWTResetPassword(secret, email, currentTime, currentTime.Add(DefaultResetPasswordExpiration))
+	email = StringValue(user.Email)
+	// TODO (@shimi9276) create template for sending the email with link for reset using the passed template
+	token, err := c.generateResetPasswordToken(email)
 	if err != nil {
-		c.Logger.WithError(err).WithField("email", email).Debug("failed to create a token")
-		writeError(w, http.StatusInternalServerError, err)
-		return
+		return err
 	}
-	// TODO (@shimi9276) create template for sending the email with link for reset
-	err = c.sendResetPasswordEmail(email, token)
+	err = c.Emailer.SendEmailWithLimit([]string{email}, token, token, nil)
 	if err != nil {
-		c.Logger.WithError(err).WithField("email", email).Warn("failed sending reset password email")
-	} else {
-		c.Logger.WithField("email", email).Info("reset password email sent")
+		return err
+	}
+	c.Logger.WithField("email", email).Info("reset password email sent")
+
+	return nil
+}
+
+func (c *Controller) ForgotPassword(w http.ResponseWriter, r *http.Request, body ForgotPasswordJSONRequestBody) {
+	err := c.resetPasswordRequest(r.Context(), body.Email)
+	if handleAPIError(w, err) {
```

Comment:
Handle API error at this point is not good as it will return an answer to the user based on err value - in this case if we fail to lookup the user it will render an error saying that - which is a bad practice for non secure route.

Reply:
Good point, fixed, now in any case (error or not) it will simply write no content